### PR TITLE
Fix isAuthenticated and other tab login/logout

### PR DESCRIPTION
**Pull request by doktordirk**
- isAuthenticated  uses values from storage
- authenticated updated on storage events and redirects if needed/desired

-new: storageChangedReload. 0/1 set to  '0' = do not redirect automatically to logout /login default pages after storage events if needed
-breaking: expiredRedirect changed to a  expiredReload 0/1
-breaking: hasDataAnalysed renamed to responseAnalysed

### CODE REVIEW

**Comment by doktordirk** [comment]
closes https://github.com/SpoonX/aurelia-authentication/issues/227

**Comment by RWOverdijk** [comment]
Must be rebased

**Comment by doktordirk** [comment]
question:
expiredReload rename to pageReloadOnExpired?
storageChangedReload rename to pageReloadOnStorageChange?
responseAnalysed acceptable naming?

File: src/authService.js [hunk 1]
```diff
@@ -67,9 +67,34 @@ export class AuthService {
...
+    PLATFORM.addEventListener('storage', this.storageEventHandler);
   }
 
   /**
+   * The handler used for storage events. Detects and handles authentication changes in other tabs/windows
+   *
+   * @param {StorageEvent}
+   */
+  storageEventHandler = event => {
+    if (event.key === this.config.storageKey) {
```
Comment (RWOverdijk) [inline src/authService.js]:
``` js
if (event.key !== this.config.storageKey) {
  return;
}
```

File: src/authService.js [hunk 2]
```diff
@@ -67,9 +67,34 @@ export class AuthService {
...
+      LogManager.getLogger('authentication').info('Stored token changed event');
+
+      let wasAuthenticated = this.authenticated;
+      this.authentication.responseAnalysed = false;
+      this.updateAuthenticated();
+
+      if (this.config.storageChangedReload) {
```
Comment (RWOverdijk) [inline src/authService.js]:
``` js
if (!this.config.storageChangedReload) {
  return;
}
```

File: src/baseConfig.js [hunk 3]
```diff
@@ -355,6 +356,9 @@ export class BaseConfig {
   set _current(_) {
     throw new Error('Setter BaseConfig._current is obsolete. Use BaseConfig directly instead.');
   }
+  set expiredRedirect(_) {
+    LogManager.getLogger('authentication').error('BaseConfig.expiredRedirect is obsolete. Use BaseConfig.expiredReload instead.');
```
Comment (RWOverdijk) [inline src/baseConfig.js]: has been removed?

Comment (RWOverdijk) [inline src/baseConfig.js]: I mean, replace obsolete by "has been removed"

**Comment by RWOverdijk** [comment]
👍

**Pull request merged**